runtime/src/lib.rs

```diff
@@ -351,10 +461,40 @@ construct_runtime!(
         DifficultyAdjustment: difficulty,
         BlockAuthor: block_author,
         Faucet: faucet,
+        EVM: pallet_evm,
+        EVMChainId: pallet_evm_chain_id,
+        BaseFee: pallet_base_fee,
+        Ethereum: pallet_ethereum,
+        ManagedAddressMapping: address_mapping,
         Contracts: pallet_contracts,
     }
 );
 
+#[derive(Clone)]
+pub struct TransactionConverter;
+
+impl fp_rpc::ConvertTransaction<UncheckedExtrinsic> for TransactionConverter {
+    fn convert_transaction(&self, transaction: pallet_ethereum::Transaction) -> UncheckedExtrinsic {
+        UncheckedExtrinsic::new_unsigned(
+            pallet_ethereum::Call::<Runtime>::transact { transaction }.into(),
+        )
+    }
+}
+
+impl fp_rpc::ConvertTransaction<opaque::UncheckedExtrinsic> for TransactionConverter {
+    fn convert_transaction(
+        &self,
+        transaction: pallet_ethereum::Transaction,
+    ) -> opaque::UncheckedExtrinsic {
+        let extrinsic = UncheckedExtrinsic::new_unsigned(
+            pallet_ethereum::Call::<Runtime>::transact { transaction }.into(),
+        );
+        let encoded = extrinsic.encode();
+        opaque::UncheckedExtrinsic::decode(&mut &encoded[..])
+            .expect("Encoded extrinsic is always valid")
+    }
+}
+
```

Comment:
I believe this is no longer necessary. See issue https://github.com/paritytech/frontier/issues/508 and PR https://github.com/paritytech/frontier/pull/559 for context.